src/mixins/date-mixin.ts

```diff
@@ -1,81 +1,159 @@
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Mixins } from 'vue-property-decorator'
+import { mapState } from 'vuex'
+import { isDate } from 'lodash'
+import { CommonMixin } from '@/mixins'
 
 /**
  * Mixin that provides some useful date utilities.
  */
-@Component({})
-export default class DateMixin extends Vue {
+@Component({
+  computed: {
+    ...mapState(['currentDate'])
+  }
+})
+export default class DateMixin extends Mixins(CommonMixin) {
+  readonly currentDate!: string
   readonly MS_IN_A_DAY = (1000 * 60 * 60 * 24)
 
   /**
-   * Converts a JavaScript date object to a simple date string.
-   * @param date The date to convert.
-   * @returns A simple date string formatted as YYYY-MM-DD.
+   * Fetches and returns the web server's current date (in UTC).
+   * Used to bypass the user's local clock/timezone.
+   * Ref: https://www.npmjs.com/package/serverdate
+   * @returns a promise to return a Date object
    */
-  dateToUsableString (date: Date): string {
-    if (!date || date.toString() === 'Invalid Date') return null
-
-    const yyyy = date.getFullYear().toString()
-    const mm = (date.getMonth() + 1).toString().padStart(2, '0')
-    const dd = date.getDate().toString().padStart(2, '0')
+  async getServerDate (): Promise<Date> {
+    const input = window.location.origin
+    const init: RequestInit = { cache: 'no-store', method: 'HEAD' }
+
+    // don't call fetch() during Jest tests
+    // because it's not defined
+    if (this.isJestRunning) return new Date()
+
+    const { headers, ok, statusText } = await fetch(input, init)
+
+    if (!ok) {
+      // eslint-disable-next-line no-console
+      console.warn('Unable to get server date - using browser date instead')
+      // fall back to local date
+      // NB: filing  may contain invalid dates/times
+      return new Date()
+    }
 
-    return `${yyyy}-${mm}-${dd}`
+    return new Date(headers.get('Date'))
   }
 
   /**
-   * Converts a date to the preferred display format
-   * @param date The date to display
-   * @returns A date in display format
+   * Converts a Date object to a date string (YYYY-MM-DD) in Pacific timezone.
+   * @example "2021-01-01 07:00:00 GMT" -> "2020-12-31"
+   * @example "2021-01-01 08:00:00 GMT" -> "2021-01-01"
+   * @example "2021-01-01 00:00:00 PST" -> "2021-01-01"
+   * @example "2021-01-01 23:59:59 PST" -> "2021-01-01"
    */
-  toDisplayDate (date: string): string {
-    const newDate = new Date(date)
-    const month = newDate.toLocaleString('default', { month: 'long' })
-    const day = newDate.getDate()
-    const year = newDate.getFullYear()
+  dateToDateString (date: Date): string {
+    // safety check
+    if (!isDate(date) || isNaN(date.getTime())) return null
+
+    const dateStr = date.toLocaleDateString('en-CA', {
+      timeZone: 'America/Vancouver'
```

Comment:
The key difference here (and in other methods) is that the timezone is now hard-coded to Pacific.

Thus, our UIs will always display in "Pacific time zone" regardless of what tz the user is actually in (or whatever they've set their browser/OS clock to).

In other words, "12:01am" always means the same thing: the first minute of the day in PST/PDT (ie, absolute terms).